client/lib/media/utils/get-file-uploader.js

```diff
@@ -7,16 +7,15 @@ const debug = debugFactory( 'calypso:media' );
 /**
  * Internal dependencies
  */
-import { reduxGetState } from 'calypso/lib/redux-bridge';
 import wpcom from 'calypso/lib/wp';
 import { getEditorPostId } from 'calypso/state/editor/selectors';
 
-export const getFileUploader = () => ( file, siteId ) => {
+export const getFileUploader = ( getState ) => ( file, siteId ) => {
```

Comment:
This works, but it's essentially a Redux thunk action implemented in an unusual way. It could be written as:
```js
const uploadFile = ( file, siteId ) => ( dispatch, getState ) => {
  const postId = getEditorPostId( getState() );
  return wpcom.site( siteId ).addMediaFile( file );
}
```
Then `addMedia` doesn't need to create `uploader` and pass it to `uploadMedia`. Instead, `uploadMedia` can simply dispatch a sub-action:
```js
await dispatch( uploadFile( file, siteId ) );
```
After this refactoring, `addMedia` and `uploadMedia` become identical.

Reply:
I agree, but this sounds like a refactor for another PR, no? The purpose of this PR is to remove `redux-bridge` and nothing more. 

I'm happy to follow-up in another PR because I agree it's not the best example of a thunk. 

What do you think?

Reply:
Follow-up PR is fine for me, just wanted to point out that the current `getFileUploader` is a non-standard pseudo-thunk and can easily be fixed :+1:

Reply:
I have second thoughts about this suggestion. It does make sense to me to keep a separate uploader function, because we have an external uploader (`getExternalUploader`), and this one (`getFileUploader`). So it kind of makes sense to keep those as separate actions.

To get around the weird pseudo-thunk pattern of `getFileUploader` we could grab the post ID and pass it to `getFileUploader` from either `addMedia` or `uploadMedia`. That would make it just a simple utility function and not connected to the state at all. Would that work for you?

Reply:
The `uploader` param could be an action creator that has `( file, siteId )` params and returns an action thunk.

They are implemented as something like:
```js
const getExternalUploader = ( service ) => ( file, siteId ) => () => {
  return wpcom.undocumented().site( siteId ).uploadExternalMedia( service, file );
}
```
and
```js
const getFileUploader = () => ( file, siteId ) => ( _, getState ) => {
  const postId = getEditorPostId( reduxGetState() );
  return wpcom.site( siteId ).addMediaFiles( { parent_id: postId, file } );
};
```
And `uploadMedia` would create and dispatch the uploader action:
```js
dispatch( uploader( file, siteId ) );
```

Would that work?

Reply:
> 1. Open Gutenframe editor and Calypso Media Modal in it.
> 2. Upload a file to the Media Modal.
> 3. Close without inserting the new image to the post. Or insert another image.
>
> Result: the uploaded image's parent ID is still set to the edited post, although it was never inserted to it.

I think this behaves the same in the WordPress core, too (since before Gutenberg, since the ages of the classic editor). I actually feel this behavior makes sense. If we're uploading an image while we're in a post context, the image should be attached to that post, regardless of whether we decide to insert it in the content or not.

Reply:
But I agree that we should pass the post ID to `addMedia` and `uploadMedia` specifically. The uploader shouldn't care about it at all. I'll draft a PR for that part.

Reply:
https://github.com/Automattic/wp-calypso/pull/53337